File: stats.py
import  sys
import os

def collect_file_stats(filename):
    lines = 0
    words = 0
    chars = 0
    try:
        with open(filename, "r", encoding="utf-8") as f:
            for line in f:
                lines += 1
                words += len(line.split())
                chars += len(line)
        return {
            "filename": filename,
            "lines": lines,
            "words": words,
            "characters": chars,
        }
    except Exception as e:
        print(f"Could not open file: {filename} ({e})")
        return None

def main():
    if len(sys.argv) < 2:
        print("Usage: python stats.py <file1> [file2 ...]")
        return
    for filepath in sys.argv[1:]:
        if not os.path.isfile(filepath):
            print(f"File not found: {filepath}")
            continue
        stats = collect_file_stats(filepath)
        if stats:
            print(f"File: {stats['filename']}")
            print(f"Lines: {stats['lines']}")
            print(f"Words: {stats['words']}")
            print(f"Characters: {stats['characters']}")

if __name__ == "__main__":
    main()

File: stats.c
#include <stdio.h>

void calculate_file_stats(const char *filename) {
    FILE *fp = fopen(filename, "r");
    if (!fp) {
        printf("Could not open file: %s\n", filename);
        return;
    }

    int ch;
    int lines = 0, words = 0, chars = 0, in_word = 0;
    while ((ch = fgetc(fp)) != EOF) {
        chars++;
        if (ch == '\n') lines++;
        if (ch == ' ' || ch == '\n' || ch == '\t') {
            in_word = 0;
        } else if (!in_word) {
            in_word = 1;
            words++;
        }
    }

    fclose(fp);

    printf("File: %s\n", filename);
    printf("Lines: %d\n", lines);
    printf("Words: %d\n", words);
    printf("Characters: %d\n", chars);
}